tidb-in-kubernetes: add doc to restart tidb components on k8s (#1759)
pingcap/docs: tidb-in-kubernetes/maintain/restart.md

---

> title: Restart TiDB Cluster in Kubernetes

Reviewer: To keep it consistent with the `Monitor a TiDB Cluster in Kubernetes` doc.
```suggestion
title: Restart a TiDB Cluster in Kubernetes
```

> summary: Learn how to restart the TiDB cluster in the Kubernetes cluster.

Reviewer: ```suggestion
summary: Learn how to restart a TiDB cluster in the Kubernetes cluster.
```

category: how-to
---

> # Restart TiDB Cluster in Kubernetes

Reviewer: ```suggestion
# Restart a TiDB Cluster in Kubernetes
```

> This document describes how to forcibly restart the TiDB cluster in the Kubernetes cluster, including restarting a Pod, restarting all Pods of a component, and restarting all Pods of the TiDB cluster.

Reviewer: ```suggestion
This document describes how to forcibly restart a TiDB cluster in the Kubernetes cluster, including restarting a Pod, restarting all Pods of a component, and restarting all Pods of the TiDB cluster.
```

> **Note:**
>
> TiDB Operator v1.0.x only supports restarting the Pod in a forcible way.
>

> > - If the PD Pod being restarted is the PD leader, its leadership does not automatically shift, causing the PD service to be momentarily unavailable.

Reviewer: - Perhaps "temporarily" is more commonly used in our doc.
- In this note, we need to put emphasis on the result, so I suggest the following changes.

```suggestion
> - If the PD Pod being restarted is the PD leader, its leadership does not automatically shift, which can cause the PD service to be temporarily unavailable.
```

> > - If the TiKV Pod being restarted contains a Region leader of the TiKV cluster, its leadership does not automatically shift, causing the requests of accessing the corresponding data to fail.

Reviewer: ```suggestion
> - If the TiKV Pod being restarted contains a Region leader of the TiKV cluster, its leadership does not automatically shift, which can cause the requests of accessing the corresponding data to fail.
```

> > - If the TiKV Pod being restarted contains a Region leader of the TiKV cluster, its leadership does not automatically shift, which can cause the requests of accessing the corresponding data to fail.

Reviewer 2: From our current documentation, usually, we say leader `transfer` instead of `shift`.

> > - Restarting the TiDB Pod causes the requests of accessing this Pod to fail.

Reviewer: Inappropriate tone, I think. To convey a kind of possibility.
```suggestion
> - Restarting the TiDB Pod can cause the requests of accessing this Pod to fail.
```

## Forcibly restart a Pod

Execute the following command to forcibly restart a Pod:

{{< copyable "shell-regular" >}}

```shell
kubectl delete pod -n <namespace> <pod-name>
```

## Forcibly restart all Pods of a component

Execute the following command to check the Pod list of a component:

{{< copyable "shell-regular" >}}

```shell
kubectl get pod -n <namespace> -l app.kubernetes.io/component=<component-name>
```

Execute the following command to forcibly restart all Pods of a component:

{{< copyable "shell-regular" >}}

```shell
kubectl delete pod -n <namespace> -l app.kubernetes.io/component=<component-name>
```

> **Note:**
>

> > - Replace `<component-name>` in the above commands with `pd`, `tidb`, or `tikv` to forcibly restart all Pods of the `PD`, `TiDB`, or `TiKV` component.

Reviewer: ```suggestion
> - To forcibly restart all Pods of the `PD`, `TiDB`, or `TiKV` component, replace `<component-name>` in the above commands with `pd`, `tidb`, or `tikv` respectively.
```
